lib/proxy_api/dhcp.rb

```diff
@@ -91,7 +91,7 @@ def set(subnet, args)
     # [+mac+]    : String in coloned sextuplet format
     # Returns    : Boolean status
     def delete(subnet, mac)
-      parse super("#{subnet}/#{mac}")
+      parse super("#{subnet}/mac/#{mac}")
     rescue RestClient::ResourceNotFound
```

Comment:
Seems like this rescue would never be reached under normal conditions, 404s are ignored and treated like "OK, it's deleted anyway" 

https://github.com/theforeman/smart-proxy/blob/develop/modules/dhcp/dhcp_api.rb#L139